lib/npg_qc/autoqc/checks/generic/artic.pm

```diff
@@ -270,9 +275,23 @@ sub _add_missing_results {
   my $master_ref = $self->inflate_rpts($self->rpt_list)->[0];
   delete $master_ref->{tag_index};
 
+  # Using the attribute that is only available for the samplesheet lims driver!
+  my $lims = st::api::lims->new(id_run   => $master_ref->{id_run},
+                                position => $master_ref->{position});
+  my $spiked_in_control_ti = $lims->spiked_phix_tag_index;
+
   for my $ti ( keys %{$self->_reads_count} ) {
-    $ti or next; # not interested in tag zero
+    # Skip tag zero.
+    $ti or next;
+    # Skip spiked-in controls (if present).
+    if (defined $spiked_in_control_ti and ($ti == $spiked_in_control_ti)) {
```

Comment:
These are numeric? In the JSON below these look like strings.

Reply:
These are integers. Might be strings in JSON as used as hash keys. Implicit conversion will be done by Perl.

Reply:
I'm a bit reluctant to let Perl do the right thing. It involves extra memory allocation, and you can't put it back into JSON the way it came out necessarily... No issue here though.